packages/headless/src/features/actions-history/ipx-actions-history-actions.ts

```diff
@@ -0,0 +1,17 @@
+import {createAsyncThunk} from '@reduxjs/toolkit';
+import {historyStore} from '../../api/analytics/coveo-analytics-utils';
+import {SearchAppState} from '../../state/search-app-state';
+
+export const addPageViewEntryInActionsHistory = createAsyncThunk(
+  'analytics/addPageViewEntry',
+  async (itemPermanentId: string, {getState}) => {
+    const state = getState() as SearchAppState;
+    if (state.configuration.analytics.enabled) {
+      historyStore.addElement({
+        name: 'PageView',
+        value: itemPermanentId,
+        time: JSON.stringify(new Date()),
+      });
+    }
+  }
+);
```

Comment:
Coupled with with other comment below, I would make this just a generic function for all entry in action history, not something just specific for pageview VS IPX.

```ts
export const addEntryInActionsHistory = createAsyncThunk(
  'analytics/actionsHistory/add',
  async (entry: HistoryElement, {getState}) => {
    const state = getState() as SearchAppState;
    if (state.configuration.analytics.enabled) {
      historyStore.addElement(entry);
    }
  }
);
```